foundations/src/telemetry/driver.rs

```diff
@@ -49,8 +48,8 @@ impl TelemetryDriver {
     ///
     /// Returns `None` if the server wasn't spawned.
     #[cfg(feature = "telemetry-server")]
-    pub fn server_addr(&self) -> Option<SocketAddr> {
-        self.server_addr
+    pub fn server_addr(&self) -> Option<String> {
```

Comment:
Why is this not a `TelemetryServerAddr`?

Reply:
`server_addr` is only used for logging, always needed as a String